src/templates/course-catalog.js

```diff
@@ -0,0 +1,271 @@
+/**
+ *  Copyright 2020 Adobe. All rights reserved.
+ *  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License. You may obtain a copy
+ *  of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License.
+ */
+
+/** @jsx jsx */
+import { Fragment, useEffect, useState } from "react"
+import { css, jsx } from "@emotion/react"
+import { graphql } from "gatsby"
+import CourseNav from "../components/CourseNav"
+import DocLayout from "../components/doclayout"
+import ExperimentalBadge from "../components/ExperimentalBadge"
+import PageActions from "../components/PageActions"
+import SiteMenu from "../components/SiteMenu"
+import RenderMdx from "../components/RenderMdx"
+
+import {
+  ActionButton,
+  Divider,
+  Flex,
+  ProgressCircle,
+  View,
+  Well
+} from "@adobe/react-spectrum"
+import {
+  ActionButtons,
+  Contributors,
+  Link,
+} from "@adobe/parliament-ui-components"
+
+// TODO: CLEAN - I am NOT proud of ANY of this
+// course.js template stores without leading slash
+const courseDir = (page) => page.path.split("README.md").shift().replace(/\/$/, "")
+const coursePages = (pages, catalogDir) => (
+  flattenPages(pages).filter((page) => {
+    const pieces = page.path.split(catalogDir).pop().split("/")
+
+    return page.path.indexOf(catalogDir) === 0 &&
+      pieces.length === 3 &&
+      pieces.pop() === "README.md"
+  })
+)
+
+const completedModules = (page, progress) =>
+  progress[page.path] && Object.keys(progress[page.path])
+        .filter((version) => progress[page.path][version])
+
+// TODO: actually match up versions
+const courseCompleted = (coursePage, progress) =>
+  coursePage.pages.length === completedModules(coursePage, progress).length
+
+const flattenPages = (pages) => {
+  if (pages === null) {
+    return []
+  }
+
+  let flat = []
+  const find = (page) => {
+    flat.push(page)
+
+    if (page.pages) {
+      page.pages.forEach(find)
+    }
+  }
+
+  pages.forEach(find)
+
+  flat = flat.flat()
+  return flat.filter(
+    (page, index) => page.path && page.path !== flat[index + 1]?.path
+  )
+}
+
+const courseButton = (page) => (
+  <ActionButton minWidth={300} margin={4}>
+    <a href={page.path}>{page.title}</a>
```

Comment:
Alright, I see why there is different behaviour. You don't want to use raw url as when the project is compile with the path prefix this href will probably be invalid. Instead use:

```
<a onClick={() => navigate(page.path)}>{page.title}</a>
```

Of course, you'd have to import it from `gatsby`

```
import { graphql, navigate } from "gatsby"
```